ledger-core/v2/log/logfmt/object.go

```diff
@@ -17,7 +18,7 @@ type LogObject interface {
 }
 
 type LogObjectMarshaller interface {
-	MarshalLogObject(LogObjectWriter, LogObjectMetricCollector) string
+	MarshalLogObject(LogObjectWriter, LogObjectMetricCollector) (msg string, defMsg bool)
```

Comment:
defMsg is not very clear naming. I'm not very clear where do you need the fact defMsg true/false

Maybe API like Msg{Text: string, TxtDefined: bool} would be more clear?

Reply:
Creating a struct is needless here as these 2 values are only returned, but are never passed together anywhere else.

defMsg means that the msg is generated by default (not provided by user).
A default msg is generated for logger / user convenience and contains struct type name.
But this is not needed for error marshalling, so the bool flag is checked inside the error marshaller to ignore msg when it is default.

This can be changed into an input parameter, but internally the message will be extracted anyway (to reuse same object marshaller for log and for errors). So it will be misleading as an input parameter also.